packages/mds-jurisdiction-service/index.ts

```diff
@@ -0,0 +1,156 @@
+/*
+    Copyright 2019-2020 City of Los Angeles.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+ */
+
+import { UUID, Timestamp, Jurisdiction } from '@mds-core/mds-types'
+import { ConnectionManager } from '@mds-core/mds-orm'
+import { NotFoundError, ServerError, ConflictError, ValidationError } from '@mds-core/mds-utils'
+import { DeepPartial } from 'typeorm'
+import { InsertReturning } from '@mds-core/mds-orm/types'
+import logger from '@mds-core/mds-logger'
+import { validateJurisdiction } from '@mds-core/mds-schema-validators'
+import { JurisdictionEntity } from './entities'
+import ormconfig from './ormconfig'
+
+import uuid = require('uuid')
+
+type JurisdictionServiceResult<TSuccess> = [Error, null] | [null, TSuccess]
+const Success = <TSuccess>(result: TSuccess): JurisdictionServiceResult<TSuccess> => [null, result]
+const Failure = <TSuccess>(error: Error): JurisdictionServiceResult<TSuccess> => [error, null]
+
+interface GetJurisdictionOptions {
+  effective: Timestamp
+}
+
+const manager = ConnectionManager(ormconfig)
+
+const initialize = async () => manager.initialize()
+
+const AsJurisdiction = (effective: Timestamp = Date.now()) => (
+  entity: JurisdictionEntity | undefined
+): Jurisdiction | null => {
+  if (entity) {
+    const { jurisdiction_id, agency_key, versions } = entity
+    const version = versions.find(properties => effective >= properties.timestamp)
+    if (version) {
+      const { agency_name, geography_id, timestamp } = version
+      if (geography_id !== null) {
+        return {
```

Comment:
I believe at one point we discussed including a stable, plain English jurisdiction identifier like 'jurisdiction/geography_name', did we decide against this?

Reply:
There is a static `agency-key` associated with a jurisdiction. I don't think we need to add a static identifier to geographies as well.

Reply:
Makes sense

Reply:
We could support getting a single jurisdiction via its `agency_key` in addition to its `jurisdiction_id` though